scripts/binary/build.ts

```diff
@@ -33,6 +37,38 @@ interface BuildCypressAppOpts {
   keepBuild?: boolean
 }
 
+/**
+ * Windows has a max path length of 260 characters. To avoid running over this when unzipping the
+ * built binary, this function attempts to guard against excessively-long paths in the binary by
+ * assuming an install default cache location and checking if final paths exceed 260 characters.
+ */
+async function checkMaxPathLength () {
+  if (process.platform !== 'win32') return log('#checkMaxPathLength (skipping since not on Windows)')
+
+  // This is the Cypress cache dir path on a vanilla Windows Server VM. We can treat this as the typical case.
+  const typicalWin32PathPrefixLength = 'C:\\Users\\Administrator\\AppData\\Local\\Cypress\\Cache\\10.0.0\\'.length
+  const maxRelPathLength = 260 - typicalWin32PathPrefixLength
+
+  log(`#checkMaxPathLength (max abs path length: ${maxRelPathLength})`)
+
+  const distDir = meta.distDir()
+  const allRelPaths = (await globAsync('**/*', { cwd: distDir, absolute: true }))
+  .map((p) => p.slice(distDir.length))
+
+  if (!allRelPaths.length) throw new Error('No binary paths found in checkMaxPathLength')
+
+  const violations = allRelPaths.filter((p) => p.length > maxRelPathLength)
+
+  if (violations.length) {
+    throw new Error([
+      `${violations.length} paths in the built binary were too long for Windows. Either hoist these files or remove them from the build.`,
```

Comment:
Does this place restrictions on the binary path length user should provide via the ENV override or this is specific to only building the binary?

Reply:
Only to building the binary, to prevent us from releasing a .zip with the most common broken scenario.

To wrap up UNIFY-1449 I intend to add a better error to the CLI when this kind of failure to unzip occurs with instructions on how to mitigate.